pydantic/config.py

```diff
@@ -36,183 +34,159 @@ class Extra(str, Enum):
-        else:
-            field_info = {}
-
-        if 'alias' in field_info:
-            field_info.setdefault('alias_priority', 2)
-
-        if field_info.get('alias_priority', 0) <= 1 and cls.alias_generator:
-            alias = cls.alias_generator(name)
-            if not isinstance(alias, str):
-                raise TypeError(f'Config.alias_generator must return str, not {alias.__class__}')
-            field_info.update(alias=alias, alias_priority=1)
-        return field_info
-
-    @classmethod
-    def prepare_field(cls, field: Any) -> None:
-        """
-        Optional hook to check or modify fields during model creation.
-        """
-        pass
-
-
-def get_config(config: Union[ConfigDict, Type[object], None]) -> Type[BaseConfig]:
+    post_init_call: Literal['before_validation', 'after_validation']  # TODO remove
+
+    # TODO where to impelement these two classmethods?
+    # @classmethod
+    # def get_field_info(cls, name: str) -> Dict[str, Any]:
+    #     """
+    #     Get properties of FieldInfo from the `fields` property of the config class.
+    #     """
+
+    #     fields_value = cls.fields.get(name)
+
+    #     if isinstance(fields_value, str):
+    #         field_info: Dict[str, Any] = {'alias': fields_value}
+    #     elif isinstance(fields_value, dict):
+    #         field_info = fields_value
+    #     else:
+    #         field_info = {}
+
+    #     if 'alias' in field_info:
+    #         field_info.setdefault('alias_priority', 2)
+
+    #     if field_info.get('alias_priority', 0) <= 1 and cls.alias_generator:
+    #         alias = cls.alias_generator(name)
+    #         if not isinstance(alias, str):
+    #             raise TypeError(f'Config.alias_generator must return str, not {alias.__class__}')
+    #         field_info.update(alias=alias, alias_priority=1)
+    #     return field_info
+
+    # @classmethod
+    # def prepare_field(cls, field: Any) -> None:
+    #     """
+    #     Optional hook to check or modify fields during model creation.
+    #     """
+    #     pass
+
+
+config_keys = set(ConfigDict.__annotations__.keys())
+
+
+def _default_config() -> ConfigDict:
+    return ConfigDict(
+        title=None,
+        str_to_lower=False,
+        str_to_upper=False,
+        str_strip_whitespace=False,
+        str_min_length=0,
+        str_max_length=None,
+        extra=Extra.ignore,
+        frozen=False,
+        populate_by_name=False,
+        use_enum_values=False,
+        validate_assignment=False,
+        arbitrary_types_allowed=False,
+        undefined_types_warning=True,
+        from_attributes=False,
+        alias_generator=None,
+        keep_untouched=(),
+        json_loads=json.loads,
+        json_dumps=json.dumps,
+        json_encoders={},
+        allow_inf_nan=True,
+        strict=False,
+        copy_on_model_validation='shallow',
+    )
+
+
+def get_config(config: Union[ConfigDict, Type[object], None] = None) -> ConfigDict:
```

Comment:
```suggestion
def get_config(config: ConfigDict | None = None) -> ConfigDict | None:
```
As per my above comment, we can keep `config_dict` partial.

Reply:
Dropping `Type[object]` does also mean removing the ability to pass a class object into dataclass decorator?

```python
@dataclass_transform(kw_only_default=True, field_specifiers=(Field, FieldInfo))
def dataclass(
    _cls: Optional[Type[_T]] = None,
    *,
    init: bool = True,
    repr: bool = True,
    eq: bool = True,
    order: bool = False,
    unsafe_hash: bool = False,
    frozen: bool = False,
    config: ConfigDict | None = None, # before config: Union[ConfigDict, Type[object], None] = None,
    validate_on_init: Optional[bool] = None,
    kw_only: bool = False,
) -> Union[Callable[[Type[_T]], 'DataclassClassOrWrapper'], 'DataclassClassOrWrapper']:
```

Reply:
Good point, thinking about this a bit more, I think it would probably be good to keep support for class defined Config everywhere, with a depreciation warning.

Reply:
I'm afraid this will make the implementation much uglier and require lots of the removed code to be added back.

Annoying I know, but I think worth it.

Reply:
What do you exactly mean? Adding back support for Config-Class in BaseModel?

Reply:
Any news here?

Reply:
Sorry for the delay, yes we'll have to add back Config Class support and raise a deprecation warning.

Sorry about this, but I think it'll help people a lot.